File: VB/13_Conversion/ConvertToOFD.vb
Imports Spire.Pdf

Namespace ConvertToOFD
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Specify the output file name for the resulting OFD document
			Dim output As String = "ConvertToOFD-result.ofd"

			' Specify the input PDF file path
			Dim input As String = "..\..\..\..\..\..\Data\ConvertToOFD.pdf"

			' Create a new PdfDocument object
			Dim pdf As New PdfDocument()

			' Load the PDF file from the specified path
			pdf.LoadFromFile(input)

			' Convert the PDF to OFD format
			pdf.SaveToFile(output, FileFormat.OFD)

			' Close the PdfDocument object
			pdf.Close()

			' Launch the odf file
			FileViewer(output)
		End Sub
		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS/13_Conversion/ConvertToOFD.cs
using Spire.Pdf;
using System;
using System.Windows.Forms;

namespace ConvertToOFD
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Input and output files path
            string output = "ConvertToOFD-result.ofd";
            string input = @"..\..\..\..\..\..\Data\ConvertToOFD.pdf";

            //Create pdf document
            PdfDocument pdf = new PdfDocument();
            pdf.LoadFromFile(input);

            //Convert pdf to ofd
            pdf.SaveToFile(output,FileFormat.OFD);

            //Launch the odf file
            FileViewer(output);
        }
        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }        
    }
}
